lib/rcov/plugin.rb

```diff
@@ -1,90 +1,182 @@
 # frozen_string_literal: false
 
-require 'open-uri'
-require 'net/http'
-require 'circle_ci_wrapper'
+require "openssl"
+require "net/http"
+require "uri"
 
 module Danger
   class DangerRcov < Plugin
-    # report will get the urls from circleCi trough circle_ci_wrapper gem
-    def report(branch_name = 'master', build_name = 'build', show_warning = true)
-      current_url, master_url = CircleCiWrapper.report_urls_by_branch(branch_name, build_name)
-
-      report_by_urls(current_url, master_url, show_warning)
+    # report is called by client Dangerfiles (f.e.: https://github.com/EdgePetrol/edge-danger/blob/master/Dangerfile#L15)
+    def report(pull_request_target_branch_name, build_job_name)
+      target_branch_coverage = find_latest_branch_coverage_report_with_job(pull_request_target_branch_name, build_job_name)
+      source_branch_coverage = find_latest_branch_coverage_report_with_job(pull_request_source_branch_name, build_job_name)
+
+      print_report_diff({
+        target: {
+          branch_name: pull_request_target_branch_name,
+          coverage: target_branch_coverage,
+        },
+        source: {
+          branch_name: pull_request_source_branch_name,
+          coverage: source_branch_coverage,
+        },
+      })
     end
 
-    def report_by_urls(current_url, master_url, show_warning = true)
-      # Get code coverage report as json from url
-      @current_report = get_report(url: current_url)
-      @master_report = get_report(url: master_url)
+    private
 
-      if show_warning && @master_report && @master_report.dig('metrics', 'covered_percent').round(2) > @current_report.dig('metrics', 'covered_percent').round(2)
-        warn("Code coverage decreased from #{@master_report.dig('metrics', 'covered_percent').round(2)}% to #{@current_report.dig('metrics', 'covered_percent').round(2)}%")
+    # As the job artifacts of the current build has already been generated and persisted
+    # by previous workflow job steps, we need to iterate through recent builds
+    # in order to find the latest build job for the specified branch.
+    #
+    # We also treat the pull request target branch (e.g., staging) and the source branch
+    # the same, because we assume the coverage report has already been generated for both of them
+    # (i.e., it has been done sometime in the past for the target branch), and we want to compare
+    # the latest coverage reports of the target branch with the pull request branch, which is
+    # possibly generated in the current job run by previous steps.
+    def find_latest_branch_coverage_report_with_job(branch_name, build_job_name)
+      page_limit = 30
+      page = 0
+      while true
```

Comment:
should this be while page <= page_limit
there are much nicer ways to do this in ruby, for example 
```
page_limit.times do |page|
```